zospy/analyses/polarization.py

```diff
@@ -0,0 +1,381 @@
+        string, it should be a full system path with '.txt' as extension, after which the file will be saved in that
+        location. Defaults to None.
+
+    Returns
+    -------
+    AnalysisResult
+        A Polarization Transmission Analysis. Next to the standard data, the raw text return obtained from the analysis
+        will be present under 'RawTextData', and the txtoutfile under 'TxtOutFile'.
+    """
+
+    analysistype = constants.Analysis.AnalysisIDM.Transmission
+
+    if cfgoutfile is None:
+        fd, cfgoutfile = mkstemp(suffix='.CFG', prefix='zospy_')
+        os.close(fd)
+        cleancfg = True
+    else:
+        if not cfgoutfile.endswith('.CFG'):
+            raise ValueError('cfgfile should end with ".CFG"')
+        cleancfg = False
+
+    if txtoutfile is None:
+        fd, txtoutfile = mkstemp(suffix='.txt', prefix='zospy_')
+        os.close(fd)
+        cleantxt = True
+    else:
+        if not txtoutfile.endswith('.txt'):
+            raise ValueError('txtfile should end with ".txt"')
+        cleantxt = False
+
+    analysis = new_analysis(oss, analysistype)
+
+    # Modify the settings file
+    analysis_settings = analysis.GetSettings()
+    analysis_settings.SaveTo(cfgoutfile)
+
+    settingsbstr = b''.join(open(cfgoutfile, 'rb').readlines())
+    settingsbarr = bytearray(settingsbstr)
+
+    # Change settings - all byte indices could only be found via reverse engineering :(
+    sampling_value = getattr(
+        constants.Analysis.SampleSizes,utils.zputils.standardize_sampling(sampling)
+        ).value__
+    settingsbarr[56] = sampling_value
+    settingsbarr[60] = int(unpolarized)
+    settingsbarr[24:32] = struct.pack('<d', jx)
+    settingsbarr[32:40] = struct.pack('<d', jy)
+    settingsbarr[40:48] = struct.pack('<d', x_phase)
+    settingsbarr[48:56] = struct.pack('<d', y_phase)
+
+    with open(cfgoutfile, 'wb') as bfile:
+        bfile.write(settingsbarr)
+
+    analysis_settings.LoadFrom(cfgoutfile)
+
+    # Run analysis
+    analysis.ApplyAndWaitForCompletion()
+
+    # Get results
+    analysis.Results.GetTextFile(txtoutfile)
+    line_list = [line for line in open(txtoutfile, 'r', encoding='utf-16-le')]
+
+    # Create output dict
+    data = AttrDict()
+
+    # Add header and sections
+    hdr = line_list[0].strip().replace('\ufeff', '')
+    data['Header'] = hdr
+
+    # Go line by line (skip first 5 lines) and sort data into data dictionary
+    ifield = 0
+    for ind, line in enumerate(line_list[5:]):
+        ind += 5
+        if 'Field Pos :' in line:
+            field = line.strip()
+            if not field in data:
+                data[field] = {}
+                ifield += 1
+                data[field]['Field number'] = ifield
+        elif 'Transmission at' in line or 'Total Transmission ' in line:
+            line_split = line.split(':')
+            data[field][re.sub(' +', ' ',line_split[0]).strip()] = float(line_split[-1])

```

Comment:
@andibarg `field` may be undefined here

Reply:
Since I needed to debug this code due to the error mentioned below, I now get how this works. Due to the structure of the output, this will work.